src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -334,7 +314,7 @@ protected virtual Task ApplyResponseGrantAsync()
 
         public virtual void SignIn(ISignInContext context)
         {
-            SignInIdentityContext = new SignInIdentityContext(context.Principal, new AuthenticationProperties(context.Properties));
+            SignInContext = new SignInContext(context.Principal, new AuthenticationProperties(context.Properties));
```

Comment:
Shouldn't SignIn and SignOut also check for an AuthScheme match before doing anything here?